packages/vhd-lib/Vhd/VhdAbstract.js

```diff
@@ -360,4 +360,42 @@ exports.VhdAbstract = class VhdAbstract {
     }
     return true
   }
+
+  async readRawData(start, length, cache) {
+    const header = this.header
+    const blockSize = header.blockSize
+    const startBlockId = Math.floor(start / blockSize)
+    const endBlockId = Math.floor((start + length) / blockSize)
+
+    const startOffset = start % blockSize
+    let copied = 0
+    const result = Buffer.alloc(length, 0)
+    for (let blockId = startBlockId; blockId <= endBlockId; blockId++) {
+      let data
+      if (this.containsBlock(blockId)) {
+        if (!cache.has(blockId)) {
+          cache.set(
+            blockId,
+            new Promise((resolve, reject) => {
+              this.readBlock(blockId)
+                .then(block => {
+                  cache.set(blockId, block.data)
```

Comment:
Unnecessary.

Reply:
when removing this part, the programm stalled when trying to read the result of `readBlock` after it was already done